pkg/scheduler/objects/application.go

```diff
@@ -142,6 +149,20 @@ func (sa *Application) SetState(state string) {
 	sa.stateMachine.SetState(state)
 }
 
+func (sa *Application) recordEvent(time time.Time, appState string, prevState string) {
```

Comment:
signature cleanup:
* time is always passed in as time.Now(), we can set it here. The nano second differences are not important.
* prevState is not used as the ordering provides the previous state already